modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeFactory.java

```diff
@@ -163,30 +161,72 @@ public Type getJavaClass(RelDataType type) {
     }
 
     /**
-     * Gets ColumnType type for given class.
+     * Gets ColumnType type for RelDataType.
      *
      * @param relType Rel type.
      * @return ColumnType type or null.
      */
-    public ColumnType columnType(RelDataType relType) {
-        assert relType != null;
+    public static ColumnType relDataTypeToColumnType(RelDataType relType) {
+        assert relType instanceof BasicSqlType
```

Comment:
Do we have tickets for unsupported types?

Reply:
https://issues.apache.org/jira/browse/IGNITE-17219
https://issues.apache.org/jira/browse/IGNITE-17298